htlcswitch/switch.go

```diff
@@ -1829,23 +1830,23 @@ func (s *Switch) RemoveLink(chanID lnwire.ChannelID) error {
 func (s *Switch) removeLink(chanID lnwire.ChannelID) error {
 	log.Infof("Removing channel link with ChannelID(%v)", chanID)
 
-	link, ok := s.linkIndex[chanID]
-	if ok {
-		s.removeLiveLink(link)
-		link.Stop()
-
-		return nil
+	link, err := s.getLink(chanID)
+	if err != nil {
+		return err
 	}
 
-	link, ok = s.pendingLinkIndex[chanID]
-	if ok {
-		delete(s.pendingLinkIndex, chanID)
-		link.Stop()
+	// Remove the channel from live link indexes.
+	delete(s.pendingLinkIndex, link.ChanID())
+	delete(s.linkIndex, link.ChanID())
+	delete(s.forwardingIndex, link.ShortChanID())
+
+	// Remove the channel from channel index.
+	peerPub := link.Peer().PubKey()
+	delete(s.interfaceIndex, peerPub)
```

Comment:
As is, this'll remove all the links for a peer, rather than just remove this particular link. Also just now realizing that `removeLinkLink` did the same thing as well.

Reply:
so disconneting one peer removes forwarding over all channels w/ that peer?